mqtt/mqtt-bridge/src/persist/loader.rs

```diff
@@ -75,10 +74,13 @@ where
                         Some((new_key, _)) if new_key == key => {
                             new_batch.pop_front();
                         }
-                        _ => panic!(
-                            "Invalid MessageLoader state: new_batch is corrupted {:?} for a key {}",
-                            new_batch, key
-                        ),
+                        _ => {
+                            return Poll::Ready(Some(Err(PersistError::Loader {
+                                key: *key,
+                                loaded: self.loaded.iter().copied().collect(),
+                                new_batch: new_batch.into_iter().collect(),
+                            })));
```

Comment:
Will this end up dumping the entirety of the publication(s) as well? If so, this could be a **big** error message in the logs. If it doesn't feel free to resolve, otherwise I think it would be nice to only have the Keys :)

Reply:
From below it looks like new_batch does include publications

Reply:
initially I thought publications may be useful, but probably you are right